rotkehlchen/tests/api/test_history_export.py

```diff
@@ -0,0 +1,152 @@
+import csv
+import os
+from http import HTTPStatus
+from pathlib import Path
+
+import requests
+
+from rotkehlchen.accounting.export.csv import FILENAME_HISTORY_EVENTS_CSV
+from rotkehlchen.db.history_events import DBHistoryEvents
+from rotkehlchen.tests.utils.api import api_url_for, assert_error_response, assert_proper_response
+from rotkehlchen.tests.utils.history import prepare_rotki_for_history_processing_test
+from rotkehlchen.tests.utils.history_base_entry import add_entries
+
+
+def assert_csv_export_response(
+        response: requests.Response,
+        csv_dir: Path,
+        expected_count: int = 6,
+        is_download: bool = False,
+):
+    """
+    Asserts that a CSV export response meets certain criteria.
+    Args:
+        response: The response object returned from the CSV export request.
+        csv_dir: The directory where the CSV files are expected to be located.
+        expected_count: The expected number of rows in the CSV files.
+    Raises:
+        AssertionError: If any of the assertions fail.
+    """
+    if is_download:
+        assert response.status_code == HTTPStatus.OK
+    else:
+        assert_proper_response(response)
+        data = response.json()
+        assert data['message'] == ''
+        assert data['result'] is True
+
+    # check the csv files were generated succesfully
+    with open(os.path.join(csv_dir, FILENAME_HISTORY_EVENTS_CSV), newline='', encoding='utf-8') as csvfile:  # noqa: E501
+        reader = csv.DictReader(csvfile)
+        count = 0
+        for row in reader:
+            # base history entry field count
+            assert len(row) >= 13
+            assert row['location'] in ('kraken', 'ethereum')
+            assert row['identifier'] is not None
```

Comment:
Can't you also just for a for loop in the attributes of the `HistoryEvents` class here?